src/app/audiologist-navigation/audiologist-navigation.component.ts

```diff
@@ -17,57 +16,61 @@ import { Utilities } from '../common/utlilities';
+  get TabsEnum() { return TabsEnum; }
+  public patientID: string = Utilities.getSessionStorage('patient-id');
+  public active: boolean = true;
+  public scale: number = 0.55;
+  public state: State = new State();
+  @ViewChild(AudiologistSummaryComponent) private summaryComponent: AudiologistSummaryComponent;
 
-    public active: boolean = true;
-    public scale: number = 0.55;
-    public recommendedTests: boolean = false;
-    public suggestedTests: boolean = false;
-    public summary: boolean = true;
-    public notes: boolean = false;
+  constructor(private router: Router) {
+  }
 
-    public onToggle() {
-      if (!this.active) {
-          this.active = true;
-          console.log('is active');
-      } else {
-        this.active = false;
-        console.log('is active');
-      }
+  public onToggle() {
+    if (!this.active) {
+      this.active = true;
+      console.log('is active');
+    } else {
+      this.active = false;
+      console.log('is active');
     }
+  }
 
-    public showRecommendedTests() {
-      this.recommendedTests = true;
-      this.suggestedTests = false;
-      this.summary = false;
-      this.notes = false;
+  public submitSurvey() {
+    if (this.summaryComponent != null) {
+      this.summaryComponent.submitSurvey();
     }
+  }
 
-    public showSuggestedTests() {
-      this.recommendedTests = false;
-      this.suggestedTests = true;
-      this.summary = false;
-      this.notes = false;
-    }
-
-    public showSummary() {
-      this.recommendedTests = false;
-      this.suggestedTests = false;
-      this.summary = true;
-      this.notes = false;
-    }
+  public clearData() {
+    // clear all patient data in memory
+    Utilities.removeItemFromSessionStorage('patient-id');
+    Utilities.removeItemFromSessionStorage('tests-data');
+    Utilities.removeItemFromSessionStorage('tfi-dataRecord');
+    Utilities.removeItemFromSessionStorage('ths-dataRecord');
+    Utilities.removeItemFromSessionStorage('ths-history');
+    Utilities.removeItemFromSessionStorage('ts-dataRecord');
+    Utilities.removeItemFromSessionStorage('ts-history');
+    Utilities.removeItemFromSessionStorage('dataFromDB');
+    this.state.determineState();
+  }
 
-    public showNotes() {
-      this.recommendedTests = false;
-      this.suggestedTests = false;
-      this.summary = false;
-      this.notes = true;
-    }
-
-    public submitSurvey() {
-      if(this.summaryComponent != null) {
-        this.summaryComponent.submitSurvey();
-      }
+  public logout() {
+    this.clearData();
+    Utilities.removeItemFromSessionStorage('audiologist-pin');
+    Utilities.removeItemFromSessionStorage('permissions');
+    this.router.navigateByUrl('/home');
+  }
 
-    }
+  // CHANGE this to take an appointment object
+  public onApptLoad(appt: Object) {
+    this.patientID = appt['id'];
+    this.summaryComponent.patientID = appt['id'];
+    this.summaryComponent.ts = appt['ts'];
```

Comment:
I think there should be some type checking for this part, creating an interface for an appointment object would fix this.
`interface AppointmentObj {
patientID: string;
.
.
.
}`

Reply:
Now that we have an Appointment class I can change it to take that. But I think we need a new ticket to change the audiologist-summary to store its data as an Appointment, and then this function can be completed to set everything correctly, and then the submitSurvey() in audiologist.summary can submit an Appointment as well.